### ISSUE TITLE
Rewrite stack trace file names of SvelteKit errors

### ISSUE THREAD

**Issue by Lms24** [getsentry/sentry#50223]
### Problem Statement

Currently, symbolicated/source-mapped stack traces of SvelteKit errors have filenames which start with a bunch of `../` directory hops:

![image (1)](https://github.com/getsentry/sentry/assets/8420481/16cc37a2-6537-4148-aeae-dd6df61a1f57)

This happens because the `sources` array in the source maps have these hops. We cannot influence the naming on the SDK side, which leaves us with two options:

- Rewrite the file names in Sentry CLI (we already do similar things in `rust-sourcemap` for other filename prefixes iirc) 
- Strip the `(../)+` prefixes in the Sentry UI 

### Solution Brainstorm

After an [internal discussion](https://sentry.slack.com/archives/C04ATL9NJ6N/p1685536328313299), we settled on option 2, as it's the faster way to get this done.

I'll take a stab at fixing this myself (unless someone who actually has a clue about react 
wants to take this on :D) but it might take a while as I'm ooo for the next two weeks. 

### Product Area

Issues

**Comment by Lms24** [comment]
I thought about this extensively and after considering changing this in the SDK or during symbolication, I think the best way forward is still to fix things in the UI and leave everything else as-is.

### Further information

**Why does this happen?** 
Some bundlers (Vite apparently) generate source maps that contain relative path prefixes in the `sources` array that point from the location of the source map to the location of the source file(s). During symbolication (if source maps are available), we assign the file name from the source map to `frame.filename` which is what we use to display the title in the UI. 

**Why not fix this in the SDK?**
- We ideally want to avoid processing/changing source maps after they're emitted from bundlers because this logic could be erroneous  whenever bundlers internally change or bundler configs are changed (e.g. within the SvelteKit framework). 
- I nevertheless tried this and ended up with:
  - messed up file paths in stack frame tooltips
  - messed up error location  
   - existing issue groups would break (not terrible but nevertheless worth avoiding) 

**Why not fix this during symbolication?**
- Basically the same reasons + increased maintenance effort for symbolicator
  - We'd need to handle yet another special case for source maps of a certain framework

**Comment by Lms24** [comment]
Closing, see https://github.com/getsentry/sentry/pull/51359#issuecomment-1611727943
